src/components/screens/monitor/delegates/withForgingStatus.js

```diff
@@ -72,35 +72,46 @@ const withForgingStatus = delegatesKey => (ChildComponent) => {
       );
       const height = blocks[0] && blocks[0].height;
       this.setState({
-        nextForgers: nextForgers.reduce((accumulator, delegate, i) => ({
-          ...accumulator,
-          [delegate.publicKey]: {
-            forgingTime: moment().add(i * 10, 'seconds'),
-            nextHeight: height + i + 1,
-          },
-        }), {}),
+        nextForgers: nextForgers.map((delegate, i) => ({
+          ...delegate,
+          forgingTime: moment().add(i * 10, 'seconds'),
+          nextHeight: height + i + 1,
+          username: delegate.username,
+        })),
       });
     }
 
     componentDidUpdate(prevProps) {
       const { latestBlocks } = this.props;
       const newBlock = latestBlocks[0] || {};
+      const forgerIndex = this.state.nextForgers.findIndex(delegate =>
+        delegate.publicKey === newBlock.generatorPublicKey);
+      const nextForgers = this.getUpdatedNextForgersList(forgerIndex, newBlock.height);
+
       if (prevProps.latestBlocks[0] && prevProps.latestBlocks[0].height < newBlock.height) {
-        this.setState({
-          nextForgers: {
-            ...this.state.nextForgers,
-            [newBlock.generatorPublicKey]: {
-              forgingTime: moment().add(voting.numberOfActiveDelegates * 10, 'seconds'),
-              nextHeight: newBlock.leight + voting.numberOfActiveDelegates,
-            },
-          },
-        });
+        this.setState({ nextForgers });
         if (newBlock.height % 101 === 1) { // to update next forgers in a new round
           this.loadNextForgers(latestBlocks);
         }
       }
     }
 
+    getUpdatedNextForgersList(forgerIndex, newBlockHeight) {
+      const nextForgers = [...this.state.nextForgers];
+
+      if (forgerIndex !== -1) {
+        const forger = {
+          ...nextForgers[forgerIndex],
+          forgingTime: moment().add(voting.numberOfActiveDelegates * 10, 'seconds'),
+          nextHeight: newBlockHeight + voting.numberOfActiveDelegates,
+        };
+        nextForgers.push(forger);
+        nextForgers.splice(0, 1);
```

Comment:
💡 `shift` would IMO be more straight-forward way to achieve the same:
```suggestion
        nextForgers.shift();
```

Reply:
Thank you for the suggestions, @slaweet. Changed